crates/llama-cpp-bindings/src/engine.cc

```diff
@@ -211,11 +249,12 @@ class TextInferenceEngineImpl : public TextInferenceEngine {
         if ((request.i_batch < i) || (request.i_batch >= (i + n_tokens))) {
           continue;
         }
+        srand(request.seed);
```

Comment:
srand itself is not thread safe, let's use std::mt19937 to create a random generator in stack.

Example: https://github.com/ggerganov/llama.cpp/blob/b46757735d30f5c6ed4f20ebeccc684e02d4f3bf/examples/main/main.cpp#L171